packages/react-cosmos-component-playground/src/components/ComponentPlayground/index.jsx

```diff
@@ -0,0 +1,222 @@
+import FixtureList from '../FixtureList';
+import WelcomeScreen from '../WelcomeScreen';
+import MissingScreen from '../MissingScreen';
+import styles from './index.less';
+
+const fixtureExists = (fixtures, component, fixture) =>
+  fixtures[component] && fixtures[component].indexOf(fixture) !== -1;
+
+export default class ComponentPlayground extends Component {
+  static defaultProps = {};
+
+  // Exclude params with default values
+  static getCleanUrlParams = params =>
+    omitBy(params, (val, key) => ComponentPlayground.defaultProps[key] === val);
+
+  state = {
+    waitingForLoader: true,
+  };
+
+  componentDidMount() {
+    window.addEventListener('message', this.onMessage, false);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('message', this.onMessage);
+  }
+
+  componentWillReceiveProps({ component, fixture }) {
+    const { waitingForLoader, fixtures } = this.state;
+
+    if (!waitingForLoader) {
+      const fixtureChanged =
+        component !== this.props.component || fixture !== this.props.fixture;
+
+      if (fixtureChanged && fixtureExists(fixtures, component, fixture)) {
+        this.loaderFrame.contentWindow.postMessage(
+          {
+            type: 'fixtureSelect',
+            component,
+            fixture,
+          },
+          '*'
+        );
+      }
+    }
+  }
+
+  onMessage = ({ data }) => {
+    const { type } = data;
+
+    if (type === 'loaderReady') {
+      this.onLoaderReady(data);
+    } else if (type === 'fixtureListUpdate') {
+      this.onFixtureListUpdate(data);
+    }
+  };
+
+  onLoaderReady({ fixtures }) {
+    const { loaderFrame } = this;
+
+    this.setState({
+      waitingForLoader: false,
+      fixtures,
+    });
+
+    const { component, fixture } = this.props;
+    if (component && fixture && fixtureExists(fixtures, component, fixture)) {
+      loaderFrame.contentWindow.postMessage(
+        {
+          type: 'fixtureSelect',
+          component,
+          fixture,
+        },
+        '*'
+      );
+    }
+  }
+
+  onFixtureListUpdate({ fixtures }) {
+    this.setState({
+      fixtures,
+    });
+  }
+
+  onUrlChange = location => {
+    this.props.router.goTo(location);
+  };
+
+  render() {
+    return (
+      <div className={styles.root}>
+        {this.renderInner()}
+      </div>
+    );
+  }
+
+  renderInner() {
+    const { fullScreen } = this.props;
+    const { waitingForLoader } = this.state;
+
+    if (waitingForLoader || fullScreen) {
+      return this.renderContent();
+    }
+
+    return [this.renderLeftNav(), this.renderContent()];
+  }
+
+  renderContent() {
+    const { loaderUri, component, fixture } = this.props;
+    const { waitingForLoader, fixtures } = this.state;
+    const isFixtureSelected = !waitingForLoader && Boolean(fixture);
+    const isMissingFixtureSelected =
+      isFixtureSelected && !fixtureExists(fixtures, component, fixture);
+
+    return (
+      <div key="loader" className={styles.loader}>
+        <StarryBg>
```

Comment:
It would be nice to not render this all the time in the background (for any loaded component). StarryBg is nice but GPU is still working in the background for no reason:
![image](https://user-images.githubusercontent.com/1254386/27409819-1632a720-56ec-11e7-9430-1abd74beb9bb.png)